### PR TITLE
feat: :sparkles: add title prop to icons

### PR DESCRIPTION
**Pull request by PhilGarb**
- adds a container for icons that includes a title prop
- includes translations for titles

### CODE REVIEW

File: nuxt/components/icon/IconAddressBook.vue [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Addressbuch
+en:
+  title: Adress Book
```
Comment (dargmuesli) [inline nuxt/components/icon/IconAddressBook.vue:26]:
```suggestion
  title: Adressbuch
en:
  title: Address book
```

File: nuxt/components/icon/IconDownload.vue [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-}
-withDefaults(defineProps<Props>(), {
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Download
```
Comment (dargmuesli) [inline nuxt/components/icon/IconDownload.vue:23]:
```suggestion
  title: Herunterladen
```

File: nuxt/components/icon/IconEnvelope.vue [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Einladungen
+en:
+  title: Invitations
```
Comment (dargmuesli) [inline nuxt/components/icon/IconEnvelope.vue:26]:
```suggestion
  title: Umschlag
en:
  title: Envelope
```

We cannot be sure this icon will only be used for invitations in the future, so we should keep it general here and overwrite it if necessary.

File: nuxt/components/icon/IconExclamationCircle.vue [hunk 4]
```diff
@@ -1,23 +1,20 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Fehlerhaft
+en:
+  title: Invalid
```
Comment (dargmuesli) [inline nuxt/components/icon/IconExclamationCircle.vue:19]:
```suggestion
  title: Erroneous
```

File: nuxt/components/icon/IconEyeOff.vue [hunk 5]
```diff
@@ -1,25 +1,26 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Privat
+en:
+  title: Private
```
Comment (dargmuesli) [inline nuxt/components/icon/IconEyeOff.vue:25]:
```suggestion
  title: Verstecken
en:
  title: Hide
```

File: nuxt/components/icon/IconInformationCircle.vue [hunk 6]
```diff
@@ -1,23 +1,20 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Ungültig
+en:
+  title: Invalid
```
Comment (dargmuesli) [inline nuxt/components/icon/IconInformationCircle.vue:19]:
```suggestion
  title: Information
en:
  title: Information
```

File: nuxt/components/icon/IconKey.vue [hunk 7]
```diff
@@ -1,25 +1,26 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Entsperren
+en:
+  title: Unlock
```
Comment (dargmuesli) [inline nuxt/components/icon/IconKey.vue:25]:
```suggestion
  title: Schlüssel
en:
  title: Key
```

File: nuxt/components/icon/IconListOl.vue [hunk 8]
```diff
@@ -1,26 +1,27 @@
...
-}
-withDefaults(defineProps<Props>(), {
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Sortierte Liste
```
Comment (dargmuesli) [inline nuxt/components/icon/IconListOl.vue:24]:
```suggestion
  title: Aufzählung
```

as recommended by @azolus.

File: nuxt/components/icon/IconListUl.vue [hunk 9]
```diff
@@ -1,26 +1,27 @@
...
-}
-withDefaults(defineProps<Props>(), {
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Unsortierte Liste
```
Comment (dargmuesli) [inline nuxt/components/icon/IconListUl.vue:24]:
```suggestion
  title: Liste
```

File: nuxt/components/icon/IconPaperPlane.vue [hunk 10]
```diff
@@ -1,26 +1,27 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Einladung
+en:
+  title: Invitation
```
Comment (dargmuesli) [inline nuxt/components/icon/IconPaperPlane.vue:26]:
```suggestion
  title: Versenden
en:
  title: Send
```

File: nuxt/components/icon/IconParagraph.vue [hunk 11]
```diff
@@ -1,26 +1,27 @@
...
-}
-withDefaults(defineProps<Props>(), {
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Paragraf
```
Comment (dargmuesli) [inline nuxt/components/icon/IconParagraph.vue:24]:
```suggestion
  title: Absatz
```

File: nuxt/components/icon/IconQrCode.vue [hunk 12]
```diff
@@ -1,25 +1,26 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: QR Code
+en:
+  title: QR Code
```
Comment (dargmuesli) [inline nuxt/components/icon/IconQrCode.vue:25]:
```suggestion
  title: QR-Code
en:
  title: QR code
```

File: nuxt/components/icon/IconSignOut.vue [hunk 13]
```diff
@@ -1,25 +1,26 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Ausloggen
+en:
+  title: Sign-out
```
Comment (dargmuesli) [inline nuxt/components/icon/IconSignOut.vue:25]:
```suggestion
  title: Ausloggen
en:
  title: Sign out
```

File: nuxt/components/icon/IconUser.vue [hunk 14]
```diff
@@ -1,25 +1,26 @@
...
-}
-withDefaults(defineProps<Props>(), {
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: User
```
Comment (dargmuesli) [inline nuxt/components/icon/IconUser.vue:23]:
```suggestion
  title: Benutzer
```

File: nuxt/components/icon/IconUserTag.vue [hunk 15]
```diff
@@ -1,26 +1,27 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: User Tag
+en:
+  title: User Tag
```
Comment (dargmuesli) [inline nuxt/components/icon/IconUserTag.vue:26]:
```suggestion
  title: Benutzer-Tag
en:
  title: User tag
```

File: nuxt/components/icon/IconTelescope.vue [hunk 16]
```diff
@@ -25,14 +25,16 @@
...
-  classes: 'h-5 md:h-6 w-5 md:w-6 shrink-0',
-})
+const { t } = useI18n()
 </script>
+
+<i18n lang="yml">
+de:
+  title: Events
+en:
+  title: Events
```
Comment (dargmuesli) [inline nuxt/components/icon/IconTelescope.vue:39]:
```suggestion
  title: Entdecken
en:
  title: Explore
```

**Review by dargmuesli — CHANGES_REQUESTED** [review]
Thanks a ton for this tedious work! :fire: 
I've left a few translations that I think fit better. It was intelligent to use texts that relate to the current usage of those icons! They might be used in different contexts in the future though, so I've suggested some general translations.

**Comment by PhilGarb** [comment]
Thank you for the quick review. I accepted your suggestions. I agree that more general translations do make sense, although there should probably be better title props passed from usage of the individual Icons. Since just hearing something like Envelope does not really mean much out of context.

**Review by dargmuesli — APPROVED** [review]

**Pull request merged**